arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/context/RaiseContext.kt

```diff
@@ -0,0 +1,132 @@
+@file:OptIn(ExperimentalTypeInference::class, ExperimentalContracts::class)
+@file:JvmMultifileClass
+@file:JvmName("RaiseContextualKt")
+
+package arrow.core.raise.context
+
+import arrow.core.Either
+import arrow.core.NonEmptyList
+import arrow.core.NonEmptySet
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.raise.Effect
+import arrow.core.raise.RaiseDSL
+import arrow.core.raise.ensure as ensureExt
+import arrow.core.raise.ensureNotNull as ensureNotNullExt
+import arrow.core.raise.withError as withErrorExt
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+import kotlin.experimental.ExperimentalTypeInference
+import kotlin.jvm.JvmMultifileClass
+import kotlin.jvm.JvmName
+
+public typealias Raise<A> = arrow.core.raise.Raise<A>
+public typealias SingletonRaise<A> = arrow.core.raise.SingletonRaise<A>
```

Comment:
We no longer need `SingletonRaise<A>` and instead we should use `Raise<A>` + `Raise<Unit>`

Reply:
In fact, this typealias can just be `public typealias SingletonRaise<A> = Raise<Unit>` to aid in migration

Reply:
`SingletonRaise` adds some additional `bind` overloads for nullable and `Option` types, so I don't think we can really do this...